apis/v1alpha1/gateway_types.go

```diff
@@ -466,56 +448,47 @@ type RouteNamespaces struct {
 
 // GatewayAddress describes an address that can be bound to a Gateway.
 type GatewayAddress struct {
-	// Type of the Address. This is either "IPAddress" or "NamedAddress".
+	// Type of the address.
 	//
 	// Support: Extended
 	//
 	// +kubebuilder:default=IPAddress
 	Type AddressType `json:"type,omitempty"`
 
-	// Value. Examples: "1.2.3.4", "128::1", "my-ip-address". Validity of the
-	// values will depend on `Type` and support by the controller.
+	// Value of the address. The validity of the values will depend
+	// on the type and support by the controller.
+	//
+	// Examples: "1.2.3.4", "128::1", "my-ip-address".
 	//
 	// +kubebuilder:validation:MinLength=1
 	// +kubebuilder:validation:MaxLength=253
 	Value string `json:"value"`
 }
 
 // AddressType defines how a network address is represented as a text string.
-// Valid AddressType values are:
-//
-// * "IPAddress": an IPv4 or IPv6 address
-// * "NamedAddress": an opaque identifier. The interpretation
-// of the name is dependent on the controller. If a NamedAddress is
-// requested but unsupported by an implementation, the controller
-// should raise the "Detached" listener status condition on the Gateway
-// with the "UnsupportedAddress" reason.
 //
-// A NamedAddress might be a cloud-dependent ID for a "static" or "elastic" IP
+// If the requested address is unsupported, the controller
+// should raise the "Detached" listener status condition on
+// the Gateway with the "UnsupportedAddress" reason.
 //
 // +kubebuilder:validation:Enum=IPAddress;NamedAddress
 type AddressType string
 
 const (
-	// IPAddressType a textual representation of a numeric IP
-	// address. IPv4 addresses must be in dotted-decimal
-	// form. IPv6 addresses must be in a standard IPv6 text
-	// representation (see RFC 5952).
-	//
-	// If the requested address is unsupported, the controller
-	// should raise the "Detached" listener status condition on
-	// the Gateway with the "UnsupportedAddress" reason.
+	// A textual representation of a numeric IP address. IPv4
+	// addresses must be in dotted-decimal form. IPv6 addresses
+	// must be in a standard IPv6 text representation
+	// (see [RFC 5952](https://tools.ietf.org/html/rfc5952)).
 	//
-	// Support: Extended.
+	// Support: Extended
 	IPAddressType AddressType = "IPAddress"
 
-	// NamedAddressType is an opaque identifier. The interpretation
-	// of the name is dependent on the controller. If a NamedAddress is
-	// requested but unsupported by an implementation, the controller
-	// should raise the "Detached" listener status condition on the Gateway
-	// with the "UnsupportedAddress" reason.
```

Comment:
Why remove the note about controller status condition here?

Reply:
it's already documented on `AddressType`